src/hexDocument.ts

```diff
@@ -183,15 +176,15 @@ export class HexDocument extends Disposable implements vscode.CustomDocument {
 	 *
 	 * These backups are used to implement hot exit.
 	 */
-	async backup(destination: vscode.Uri, cancellation: vscode.CancellationToken): Promise<vscode.CustomDocumentBackup> {
-		await this.saveAs(destination, cancellation);
-		await vscode.workspace.fs.writeFile(vscode.Uri.parse(destination.path + ".json"), Buffer.from(JSON.stringify(this.unsavedEdits), "utf-8"));
+	async backup(destination: vscode.Uri): Promise<vscode.CustomDocumentBackup> {
+		const backup = new Backup(destination);
+		await backup.write(this.model.unsavedEdits);
```

Comment:
So are backups no longer cancellable?

Reply:
`vscode.workspace.fs.writeFile` is not cancellable 🙁